src/swap/components/Swap.tsx

```diff
@@ -34,91 +30,65 @@ export function Swap({ address, children, title = 'Swap' }: SwapReact) {
     isToQuoteLoading: false,
   });
 
-  // returns ETH balance
-  const ethBalanceResponse: UseBalanceReturnType = useBalance({
-    address,
-  });
-
-  // returns erc20 token balance
-  const fromTokenBalanceResponse: UseReadContractReturnType = useReadContract({
-    abi: erc20Abi,
-    address: fromToken?.address as Address,
-    functionName: 'balanceOf',
-    args: [address],
-    query: {
-      enabled: !!fromToken?.address && !!address,
-    },
-  });
+  const {
+    convertedBalance: convertedETHBalance,
+    roundedBalance: roundedETHBalance,
+    error: ethBalanceError,
+  } = useGetETHBalance(address);
 
-  // returns erc20 token balance
-  const toTokenBalanceResponse: UseReadContractReturnType = useReadContract({
-    abi: erc20Abi,
-    address: toToken?.address as Address,
-    functionName: 'balanceOf',
-    args: [address],
-    query: {
-      enabled: !!toToken?.address && !!address,
-    },
-  });
+  const {
+    convertedBalance: convertedFromBalance,
+    error: fromBalanceError,
+    roundedBalance: roundedFromBalance,
+  } = useGetTokenBalance(address, fromToken);
 
-  const { fromTokenBalanceError, toTokenBalanceError } = useMemo(() => {
-    const fromTokenBalanceError = getTokenBalanceErrorState({
-      token: fromToken,
-      ethBalance: ethBalanceResponse,
-      tokenBalance: fromTokenBalanceResponse,
-    });
-    const toTokenBalanceError = getTokenBalanceErrorState({
-      token: toToken,
-      ethBalance: ethBalanceResponse,
-      tokenBalance: toTokenBalanceResponse,
-    });
-    return {
-      fromTokenBalanceError,
-      toTokenBalanceError,
-    };
-  }, [
-    ethBalanceResponse,
-    fromToken,
-    fromTokenBalanceResponse,
-    toToken,
-    toTokenBalanceResponse,
-  ]);
+  const {
+    convertedBalance: convertedToBalance,
+    roundedBalance: roundedToBalance,
+    error: toBalanceError,
+  } = useGetTokenBalance(address, toToken);
 
+  /* istanbul ignore next */
```

Comment:
the coverage report was getting upset about these lines so i added the ignore statement here. i'm not sure of a way to test those values since they don't render anywhere but let me know if someone has an idea of how to test these lines. 
![Screenshot 2024-06-21 at 1 11 34 PM](https://github.com/coinbase/onchainkit/assets/170486079/3006f2f4-634d-4347-9c14-1d445f9c9de2)